### DISCUSSION TITLE
SCNetworkReachability API deprecations.

### DISCUSSION THREAD

**Discussion by FONickReichard** [Alamofire/Alamofire#3852] [General]
Found out today several of the methods from the SCNetworkReachability API collection are being marked as deprecated starting with iOS 17.4.
https://developer.apple.com/documentation/systemconfiguration/scnetworkreachability-g7d 

Based on this [WWDC video ](https://developer.apple.com/videos/play/tech-talks/111378/) it looks like [waitsForConnectivity](https://developer.apple.com/documentation/foundation/urlsessionconfiguration/2908812-waitsforconnectivity/) is a suitable replacement. 

Maybe something like this could help config setup in the [URLSessionConfiguration+Alamofire.swift](https://github.com/Alamofire/Alamofire/blob/9e9943fbdffbe9d5bb0324f16b44873bc11e6dd9/Source/Extensions/URLSessionConfiguration%2BAlamofire.swift#L2) :

```   
public static func default(waitsForConnectivity: Bool = false) -> URLSessionConfiguration {
        let configuration = URLSessionConfiguration.default
        configuration.headers = .default
        configuration.waitsForConnectivity = waitsForConnectivity
        return configuration
    }
```

Wanted to pass along this *deprecation notice as it seems to be new & so there's more time to plan it out. 

Best 

@jshier

**Comment by jshier** [comment]
Thanks for the heads up! `waitsForConnectivity` isn't really what people should've been using reachability for, as Apple has long said it shouldn't be used as a precheck for requests. A true replacement is `NWPathMonitor`, which has a similar purpose but not quite the same behavior. Rather than any sort of reachability check, the path monitor could be used as part of a `RequestRetrier` to monitor the base `URL` server and attempt retry when the path is available again. I don't anticipate shipping a direct replacement for Alamofire's reachability monitor.

**Reply by FONickReichard** [reply]
Sounds good. Thanks for the correction & heads up!

**Reply by Vyazovoy** [reply]
@jshier, may I ask to clarify you vision. Am I right understand that you intentionally don't want to migrate the reachability monitor implementations from `Reachability` to `NWPathMonitor`? Another option would be to provide a new "monitor" implemented with the `NWPathMonitor` and deprecate the old/current one.

**Reply by jshier** [reply]
`NWPathMonitor` is so much simpler than the older reachability API there's really very little value to Alamofire providing the equivalent wrapper. Now, if someone wanted to put together a `RequestRetrier` that uses a path monitor to try and wait for connectivity to come back to a failed request's domain, that might be useful, if it can be made reliable. I would gladly review such a PR. But it's not something I care to implement myself.
